src/librustc_typeck/check/pat.rs

```diff
@@ -697,15 +714,25 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             res.descr(),
             fields.len(),
             fields_ending,
-        )
-            .span_label(pat_span, format!(
+        );
+        err.span_label(pat_span, format!(
                 "expected {} field{}, found {}",
                 fields.len(),
                 fields_ending,
                 subpats.len(),
             ))
-            .span_label(res_span, format!("{} defined here", res.descr()))
-            .emit();
+            .span_label(res_span, format!("{} defined here", res.descr()));
+
+        if missing_parenthesis {
+            err.multipart_suggestion(
+                &missing_parenthesis_error_msg,
```

Comment:
Given that `missing_parenthesis_error_msg` is not used anywhere else, I would just write the literal here. Also, I do not use title capitalization, use leading lowercase: `"missing parenthesis"`

Reply:
Done